File: client/clojure/src/cdisc_library_api/specs/ct_package_codelists.clj
(ns cdisc-library-api.specs.ct-package-codelists
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.ct-package-codelists-links :refer :all]
            )
  (:import (java.io File)))


(def ct-package-codelists-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :source) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :_links) ct-package-codelists-links-spec
   })

(def ct-package-codelists-spec
  (ds/spec
    {:name ::ct-package-codelists
     :spec ct-package-codelists-data}))

File: client/scalaz/src/main/scala/org/openapitools/client/api/CtPackageCodelists.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CtPackageCodelists._

case class CtPackageCodelists (
  name: Option[String],
label: Option[String],
description: Option[String],
source: Option[String],
effectiveDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
links: Option[CtPackageCodelistsLinks])

object CtPackageCodelists {
  import DateTimeCodecs._

  implicit val CtPackageCodelistsCodecJson: CodecJson[CtPackageCodelists] = CodecJson.derive[CtPackageCodelists]
  implicit val CtPackageCodelistsDecoder: EntityDecoder[CtPackageCodelists] = jsonOf[CtPackageCodelists]
  implicit val CtPackageCodelistsEncoder: EntityEncoder[CtPackageCodelists] = jsonEncoderOf[CtPackageCodelists]
}
